asgi_rabbitmq/core.py

```diff
@@ -236,123 +203,39 @@ def publish_properties(self, channel=None):
 
     # Receive.
 
-    def receive(self, future, channels, block):
+    def receive(self, future, channel):
         """Initiate message receive."""
 
-        # Declare necessary queues in parallel.
-        queues = set(map(self.get_queue_name, channels))
-        known_queues = set()
-        queues_declared = partial(self.queues_declared, future, queues,
-                                  known_queues, channels, block)
-        for queue in queues:
-            self.amqp_channel.queue_declare(
-                queues_declared,
-                queue,
-                arguments=self.queue_arguments,
-            )
-
-    def queues_declared(self, future, queues, known_queues, channels, block,
-                        method_frame):
-        """One queue was declared.  Continue if it was the last one."""
-
-        known_queues.add(method_frame.method.queue)
-        # If all queues are known at this moment, that basically means
-        # we are in the last callback and can safely go further.  If
-        # any queue isn't known, we simply skip processing at this
-        # point.
-        unknown_queues = queues - known_queues
-        if unknown_queues:
-            return
-        if block:
-            self.start_blocking_receive(future, channels)
-        else:
-            channels = list(channels)  # Daphne sometimes pass dict.keys()
-            self.start_non_blocking_receive(future, channels)
-
-    # Blocking receive.
-
-    def start_blocking_receive(self, future, channels):
-        """Create parallel consumers in one AMQP channel."""
-
-        # In blocking mode create consumers in parallel.
-        consumer_tags = {}
-        for channel in channels:
-            tag = self.amqp_channel.basic_consume(
-                partial(self.consume_message, future, consumer_tags),
-                queue=self.get_queue_name(channel),
-            )
-            consumer_tags[tag] = channel
-        future.add_done_callback(
-            partial(self.cancel_consuming, consumer_tags=consumer_tags)
+        queue = self.get_queue_name(channel)
+        self.amqp_channel.queue_declare(
+            partial(self.handle_receive, future, channel),
+            queue,
+            arguments=self.queue_arguments,
         )
+        future.add_done_callback(partial(self.cancel_consuming, channel))
```

Comment:
this callback should be added after calling basic_consume, not after declaring the queue